### ISSUE TITLE
Too many failures testing on local but not in TravisCI

### ISSUE THREAD

**Issue by miceno** [collectiveaccess/providence#400]
Hi 

I am testing the latest latest develop branch on my local computer (macosx) using php 7.3.13, and it breaks the tests :( The report says there 149 failed tests.

The point is that I had a look at TravisCI and the builds are ok, so I thought it was something on my environment.

I debug a bit and I discovered a couple of things that look like the cause of some of the failed tests, but what it is really puzzling me is why those tests didn't fail in TravisCI :(

In fact, this is the way I found the timeout issue on the AAT curl requests, because the test failed locally.

For example, I found that `ConfigurationUpdateTest::testDeleteList` failed because there was a missing DB transaction commit here before `continue` statement: 
https://github.com/collectiveaccess/providence/blob/d99ab18a516c87c486314cf0f7a2425fd784b0ba/install/inc/Installer.php#L590-L593

But I think that failure should also show up on TravisCI, but it didn't. Weird!

I created a clean environment cloning develop branch on a new folder, and run it again, and same failures...

I run it also on PHP 7.2.26, and it also raised the around 140 failures.

I am attaching the test report:
[Providence develop - ALL_tests.html.zip](https://github.com/collectiveaccess/providence/files/4524488/Providence.develop.-.ALL_tests.html.zip)

I now I can run tests on TravisCI, but I prefer running them locally, it is faster and less time consuming.

Any suggestion on how to deal with the problem?

**Comment by collectiveaccess** [comment]
Did you run this on a clean install of the test profile?

**Comment by miceno** [comment]
Yes, I did. I remove the database, I create it again, and I also clean the app/tmp/collectiveaccess cache folder

**Comment by collectiveaccess** [comment]
And you're sure it's actually using that database? Many of the errors imply you may not be. Also, how are you running these tests?

**Comment by miceno** [comment]
Yes, it is using the testing database. I am debugging it and the testing database changes.

I am running the tests from PHPStorm, here is the command line shown:

/usr/local/Cellar/php@7.3/7.3.13/bin/php providence/vendor/phpunit/phpunit/phpunit --configuration providence/tests/phpunit.xml --teamcity

**Comment by collectiveaccess** [comment]
What version of PHPUnit do you have installed? I'm running under 7.3 as well...

**Comment by miceno** [comment]
PHPUnit 8.5.0 by Sebastian Bergmann and contributors.

**Comment by miceno** [comment]
Let me try it from the command line, without  PHPStorm IDE... I will report back...

**Comment by miceno** [comment]
Running from command line it worked!!! I don't know why it is failing in PHPStorm, I think it is something related to PHP modules available, who knows :(

**Comment by collectiveaccess** [comment]
Well that's very odd... it would be good to know why.

**Comment by miceno** [comment]
Yes, I think there is something on the way PHPStorm manage PHP modules or memory, but I didn't find anything yet. I will run from command line.

I will give it a try also on the vagrant machine.

**Comment by miceno** [comment]
I solved the issue... the problem was that I was not calling `phpunit` like it is called on TravisCI scripts:

I was running command line 👎 :

       $providence> php vendor/phpunit/phpunit/phpunit -c tests/phpunit.xml

Right command line 👍 : 

       $providence> cd tests
       $providence/tests> ../vendor/phpunit/phpunit/phpunit

This is the way it was called in TravisCI scripts.

The configuration for PHPStorm is:

- Create a custom Run/Debug configuration.
- Choose "Test Runner > Defined in configuration file"
- Set "Command Line > Custom Working Directory" to the `tests` folder.

I also realized that `phpunit.xml` does not include a test suite for refineries. Should I add it?

**Comment by miceno** [comment]
I finally dig it more into the cause of the errors, and there were several issues:

* I was running all the tests, even those that were excluded by the configuration, like those on `tests/install/ConfigurationUpdateTest.php`. I figured out to solve some of the fails, I will issue a PR later.
* Once an error or test failure arised, the database was in an inconsistent status, and the next tests failed in cascade.
* Tests to Getty external services like AAT, TGN, etc failed due to a timeout. There is a merged PR with improvements for AAT, I will extend it to all the Getty services (TGN, ULAN, etc.).
* Wikipedia lookups are failing randomly on my network (I think it is a network issue, but I think I saw it also on TravisCI). Tests following those tests also failed. I think the way to solve it in this case is "silently" marking the test as incomplete. I am preparing a PR about it and we may discuss if there are other approaches.
